src/components/Shared/GuestAuthenticationDialog.js

```diff
@@ -0,0 +1,227 @@
+      full_name: "Guest User",
+      preferred_name: "Guest",
+      email,
+      community_id: community?.id,
+      accepts_terms_and_conditions: false,
+      is_vendor: false,
+      is_guest: true,
+    };
+    setLoading(true);
+
+    apiCall("/users.create", data)
+      .then((response) => {
+        setLoading(false);
+        if (!response.success) return setError(response?.error?.message);
+        const user = response.data;
+
+        const userExistsAndIsNotAGuest =
+          user?.user_info?.user_type !== GUEST_USER;
+        if (userExistsAndIsNotAGuest) return setNotGuest(true);
+        
+        putUserInRedux(response.data);
+        localStorage.setItem(GUEST_USER_KEY, email);
+        setGuestAuthIsDone(true);
+        callback && callback(true);
+
+        setTimeout(() => {
+          close && close();
+          window.location.reload();
+        }, 1500); // close modal after 1.5 seconds
+      })
+      .catch((e) => {
+        setLoading(false);
+        console.log("GUEST_AUTHENTICATION_ERROR", e);
+      });
+  };
+
+  useEffect(() => {}, [user]);
+
+  const whenUserTypes = (e) => {
+    if (ifEnterKeyIsPressed(e)) return authenticateGuest();
+  };
+
+  const goToMainAuthPage = () => {
+    close && close();
+    history.push(links?.signin);
+  };
+
+  if (guestAuthIsDone) return <CongratulatoryMessage />;
+  return (
+    <>
+      <div className="guest-dialog-container">
+        <div>
+          <p className="responsive-p">
+            {!proceedAsGuest ? (
+              <span>
+                Hi there, please choose one of the options below to continue
+              </span>
+            ) : (
+              <span>
+                Please provide an
+                <span
+                  style={{ color: "var(--app-theme-orange)", marginLeft: 5 }}
+                >
+                  email
+                </span>{" "}
+                below. That's all you would have to do for now.
+              </span>
+            )}
+          </p>
+
+          {proceedAsGuest && (
+            <METextField
+              placeholder="example@gmail.com"
+              onChange={(e) => setEmail(e.target.value)}
+              genericProps={{ onKeyUp: whenUserTypes }}
+            />
+          )}
+
+          {error && (
+            <small style={{ color: "maroon", marginTop: 5 }}>{error}</small>
+          )}
+          {notGuest && (
+            <small
+              style={{
+                color: "maroon",
+                marginTop: 5,
+              }}
+            >
+              Hi there, it looks like you have an account with us
+              already.
+              <Link
+                to={links?.signin}
+                onClick={goToMainAuthPage}
+                style={{ textDecoration: "underline", color: "maroon" }}
+              >
+                Try signing in with this email on our sign in page.
+              </Link>{" "}
+              Or use a different email to sign in as a guest!
```

Comment:
I'm eliminating this sentence because we don't want to encourage people to use multiple emails to record stuff.  That would be a mess.